### PR TITLE
[frontendproxy] - add envoy access logs

### PR DESCRIPTION
**Pull request by puckpuck**
# Changes

Supersedes #1751 

This adds Envoy access logs to the OTLP logs pipeline. Semantic conventions were followed as much as possible for the various attributes added.

## Merge Requirements

For new feature contributions, please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [ ] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

**Review by cedricziel — APPROVED** [review]

File: src/frontendproxy/envoy.tmpl.yaml [hunk 1]
```diff
@@ -65,6 +65,94 @@ static_resources:
...
+                    typed_config:
+                      "@type": "type.googleapis.com/envoy.extensions.access_loggers.open_telemetry.v3.OpenTelemetryAccessLogConfig"
+                      common_config:
+                        log_name: "otel_envoy_accesslog"
+                        grpc_service:
+                          envoy_grpc:
+                            cluster_name: opentelemetry_collector_grpc
+                        transport_api_version: "V3"
+                      body:
+                        string_value: "[%START_TIME%] \"%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%\" %RESPONSE_CODE% %RESPONSE_FLAGS% %RESPONSE_CODE_DETAILS% %CONNECTION_TERMINATION_DETAILS% \"%UPSTREAM_TRANSPORT_FAILURE_REASON%\" %BYTES_RECEIVED% %BYTES_SENT% %DURATION% %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)% \"%REQ(X-FORWARDED-FOR)%\" \"%REQ(USER-AGENT)%\" \"%REQ(X-REQUEST-ID)%\" \"%REQ(:AUTHORITY)%\" \"%UPSTREAM_HOST%\" %UPSTREAM_CLUSTER% %UPSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_REMOTE_ADDRESS% %REQUESTED_SERVER_NAME% %ROUTE_NAME%\n"
```
Comment (cedricziel) [inline src/frontendproxy/envoy.tmpl.yaml:79]:
```suggestion
                        # yamllint disable-line rule:line-length
                        string_value: "[%START_TIME%] \"%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%\" %RESPONSE_CODE% %RESPONSE_FLAGS% %RESPONSE_CODE_DETAILS% %CONNECTION_TERMINATION_DETAILS% \"%UPSTREAM_TRANSPORT_FAILURE_REASON%\" %BYTES_RECEIVED% %BYTES_SENT% %DURATION% %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)% \"%REQ(X-FORWARDED-FOR)%\" \"%REQ(USER-AGENT)%\" \"%REQ(X-REQUEST-ID)%\" \"%REQ(:AUTHORITY)%\" \"%UPSTREAM_HOST%\" %UPSTREAM_CLUSTER% %UPSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_REMOTE_ADDRESS% %REQUESTED_SERVER_NAME% %ROUTE_NAME%\n"
```

**Comment by julianocosta89** [comment]
I'm getting an error on this one:
image

**Comment by julianocosta89** [comment]
@puckpuck are you also seeing the error I mentioned above?

**Comment by puckpuck** [comment]
It took me a while to figure this one out, but it seems like the log volume and log size is too big for the Grafana <-> OpenSearch exchange. 

I found other minor issues and optimizations as well, but will ultimately impose a limit on the # of entries that are returned to that log panel. I may settle on 100 entries since this is meant to be a demo (and update the panel title accordingly as well).

I'll update this PR later today with the fixes.

**Comment by puckpuck** [comment]
@julianocosta89 can you try again now?

**Comment by puckpuck** [comment]
> @puckpuck not getting the error, but not getting any log for `frontend-proxy`. All other services logs are working fine.
> 
> I've noticed some error messages on the collector:
> 
> ```
> otel-col                 | 2024-11-18T22:31:02.080Z	warn	batchprocessor@v0.113.0/batch_processor.go:263	Sender failed	{
> "kind": "processor", 
> "name": "batch",
> "pipeline": "logs",
> "error": "not retryable error: Permanent error: {
>     \"type\":\"mapper_parsing_exception\",
>     \"reason\":\"Could not dynamically add mapping for field [http.connection.id]. Existing mapping for [attributes.http] must be of type object but found [text].\",
>     \"caused_by\":{
>         \"type\":\"\",
>         \"reason\":\"\",
>         \"caused_by\":null
>     }
> }
> ```

Can you recheck your configuration including `otelcol-config-extras.yml` on this one? I just ran the following commands, and everything worked as expected.

```
docker system prune -af
docker compose build frontendproxy
make start
```

File: src/grafana/provisioning/dashboards/demo/demo-dashboard.json [hunk 2]
```diff
@@ -1039,6 +1039,6 @@
   "timezone": "",
   "title": "Demo Dashboard",
   "uid": "W2gX2zHVk",
-  "version": 2,
+  "version": 1,
```
Comment (julianocosta89) [inline src/grafana/provisioning/dashboards/demo/demo-dashboard.json:1042]: any reason why you are setting version to 1?

**Review by julianocosta89 — APPROVED** [review]
LGTM 🚀 
Also, thx @martinjt!

**Comment by julianocosta89** [comment]
One thing that I missed was the log level, but couldn't find a way to have that in Envoy

File: src/grafana/provisioning/dashboards/demo/demo-dashboard.json (hunk 2 above)
Comment (puckpuck) [reply src/grafana/provisioning/dashboards/demo/demo-dashboard.json:1074]: I didn't set it this specifically. The Grafana UI spit out the dashboard JSON.

**Pull request merged**